framework/src/dgkernels/DGKernelBase.C

```diff
@@ -109,53 +112,112 @@ DGKernelBase::DGKernelBase(const InputParameters & parameters)
     _save_in_strings(parameters.get<std::vector<AuxVariableName>>("save_in")),
     _diag_save_in_strings(parameters.get<std::vector<AuxVariableName>>("diag_save_in"))
 {
+  // Gather information on broken boundaries
+  auto & bnd = isParamValid("exclude_boundary")
+                   ? getParam<std::vector<BoundaryName>>("exclude_boundary")
+                   : std::vector<BoundaryName>(0);
+  auto bnd_ids = _mesh.getBoundaryIDs(bnd);
+
+  // check if the broken boundary ids are valid
+  auto & valid_ids = _mesh.meshSidesetIds();
+  std::vector<BoundaryID> diff;
+  std::set_difference(
+      bnd_ids.begin(), bnd_ids.end(), valid_ids.begin(), valid_ids.end(), std::back_inserter(diff));
+  if (!diff.empty())
+  {
+    std::ostringstream msg;
+    msg << "DGKernel '" << name()
+        << "' contains the following boundary ids that do not exist on the mesh:";
+    for (const auto & id : diff)
+      msg << " " << id;
```

Comment:
minor: We have a shortcut for this logic now if you want to use that. Check out the templated `stringify` method in `Conversion.h`